### PR TITLE
Continuation of JWKS support from PR #71

### PR DESCRIPTION
**Pull request by davidallendj**
This PR continues the work from PR #71 to add JWKS support, but removes the dynamic verifier as requested.

### CODE REVIEW

**Comment by davidallendj** [comment]
@pkieltyka Can I get a review for this?

File: jwtauth.go [hunk 1]
```diff
@@ -120,6 +137,10 @@ func VerifyToken(ja *JWTAuth, tokenString string) (jwt.Token, error) {
 }
 
 func (ja *JWTAuth) Encode(claims map[string]interface{}) (t jwt.Token, tokenString string, err error) {
+	if ja.keySet != nil {
+		return nil, "", fmt.Errorf("encode not supported")
```
Comment (alexlovelltroy) [inline jwtauth.go:141]: I'm working on rebasing and adding more documentation.  I don't understand why encode isn't supported in this case and I think we need a better error message for it.  Can you explain what this is for?

Comment (davidallendj) [reply jwtauth.go:141]: I think that's the wrong error message probably from some copy-pasta. I think the message should just reflect that the JWKS isn't set so nothing can be encoded here.

Comment (alexlovelltroy) [reply jwtauth.go:141]: Based on my understanding, I'm considering updating Encode like this:

```go

Comment (davidallendj) [reply jwtauth.go:141]: Looks fine to me. I was reading the code wrong and was thinking the `ja.keySet` was an error. I'll have to go back and look at the entire PR to recall the context why this was done.

Comment (alexlovelltroy) [reply jwtauth.go:141]: I'm refactoring it a bit now that I understand it better.

**Pull request closed without merging**